src/connection_string.ts

```diff
@@ -289,12 +286,6 @@ export function parseOptions(
       throw new MongoAPIError('URI cannot contain options with no value');
     }
 
-    if (key.toLowerCase() === 'serverapi') {
-      throw new MongoParseError(
-        'URI cannot contain `serverApi`, it can only be passed to the client'
-      );
-    }
-
     if (key.toLowerCase() === 'authsource' && urlOptions.has('authSource')) {
       // If authSource is an explicit key in the urlOptions we need to remove the implicit dbName
       urlOptions.delete('authSource');
```

Comment:
Is this a bug? Just noticed it says dbName needs to be removed, and that's also the only thing actually set in the code above, so deleting authSource does nothing

Reply:
Hmm this code is strange but I'm also not sure what it is trying to accomplish, I reviewed the specs and I don't see where dbName and authSource would conflict with one another. and implicit `dbName` implies to me refering to how dbName defaults to `test` but that doesn't happen before this point. 

I'm going to try deleting it, lets see what the waterfall says, but we can keep it and file a ticket.

Reply:
Deleting it shouldn't do anything, but I meant it should maybe be updated to delete 'dbName' instead, because of the code above that does:
` if (url.pathname !== '/' && url.pathname !== '') {
    const dbName = decodeURIComponent(
      url.pathname[0] === '/' ? url.pathname.slice(1) : url.pathname
    );
    if (dbName) {
      urlOptions.set('dbName', [dbName]);
    }
  }`
Clearly the behavior isn't tested or there would be failures.